### ISSUE TITLE
Enhance bibliography numbering

### ISSUE THREAD

**Issue by getreu** [asciidoctor/asciidoctor#1293]
For short articles I prefer this [1] kind of references. I found a nice workaround that enumerates the items:

``` asciidoc
{counter2:bib:0}

[horizontal]
[[buterin2014,{counter:bib}]][{bib}]:: 
Buterin, Vitalik. “Ethereum: A Next-Generation Smart Contract and 
Decentralized Application Platform.”, 2014.
https://github.com/ethereum/wiki/wiki/White-Paper.
```

Unfortunately this works only with the docbook backend. The html backend does not allow substitutions like `{counter:bib}`.

**Comment by mojavelinux** [comment]
What you posted should work in the HTML backend. It seems to work fine for me at least (using Asciidoctor 1.5.3 dev).

For more advanced bibliography layouts, I recommend checking out asciidoc-bib. I'd like something similar in core eventually, but at the moment I don't have time to focus on it.

Remember that you should try to avoid thinking about formatting of output when encoding content into AsciiDoc. How the data is displayed can be controlled by the backend (a custom one if necessary). The way it displays by default is really just a suggestion. You can and should customize the backend to get exactly the output you want without having to do a lot of fixing of the input...unless the input is not organized in a way that you can pick out the parts of a component.

**Comment by mojavelinux** [comment]
> For short articles I prefer this [1] kind of references.

This will be possible, at least in part, using Asciidoctor 1.5.6 thanks to the fix to #1562.

**Comment by ronaldtse** [comment]
@getreu or you could also try out [`asciidoctor-bibilography`](https://github.com/riboseinc/asciidoctor-bibliography), which allows full customization of cite keys and bridges all CSL styles. However it does require your bibliography to be in BibTeX format.

To use the `[1]` style just add this to your header (given that you have set up a BibTeX file and asciidoctor-bibilography):
```
:bibliography-style: ieee
:bibliography-tex-style: numeric
```

You could even sort the bibliography according to citation attributes or appearance in the document.

The caveat is there is currently a bug in the bibliography where the `[1]` runs into the author name with a missing space, but it should be fixed soon https://github.com/riboseinc/asciidoctor-bibliography/issues/16.
